src/lib/salad/heap.h

```diff
@@ -389,7 +389,7 @@ HEAP(reserve)(heap_t *heap)
 	void *harr = realloc(heap->harr, sizeof(struct heap_node *) * capacity);
 	if (harr == NULL)
 		return -1;
-	heap->harr = harr;
+	heap->harr = (struct heap_node **)harr;
```

Comment:
Please drop this change, seems unrelated to the patch.

Reply:
As soon as there is a compiler link sequence of box.cc -> vy_lsm.h -> vy_range.h -> salad/heap.h and C++ is on duty there occurs a compiler error: invalid conversion from ‘void*’ to ‘heap_node**’ [-fpermissive].
Can't drop the change.